code/modules/power/apc.dm

```diff
@@ -839,46 +839,48 @@
 		var/mob/living/carbon/human/H = user
 		var/datum/species/ethereal/E = H.dna.species
 		var/charge_limit = ETHEREAL_CHARGE_DANGEROUS - APC_POWER_GAIN
-		if(H.combat_mode && E.drain_time < world.time)
-			if(LAZYACCESS(modifiers, RIGHT_CLICK)) //Disarm
-				if(cell.charge == cell.maxcharge)
-					to_chat(H, "<span class='warning'>The APC is full!</span>")
+		if((E.drain_time < world.time) && LAZYACCESS(modifiers, RIGHT_CLICK))
+			if(H.combat_mode)
+				if(cell.charge <= (cell.maxcharge / 2)) // ethereals can't drain APCs under half charge, this is so that they are forced to look to alternative power sources if the station is running low
+					to_chat(H, "<span class='warning'>The APC's syphon safeties prevent you from draining power!</span>")
 					return
 				var/obj/item/organ/stomach/ethereal/stomach = H.getorganslot(ORGAN_SLOT_STOMACH)
-				if(stomach.crystal_charge < 10)
+				if(stomach.crystal_charge > charge_limit)
```

Comment:
Could you add a check somewhere to handle ethereals without their stomachs? It's going to happen at some point.

Reply:
Does `var/obj/item/organ/stomach/ethereal/stomach = H.getorganslot(ORGAN_SLOT_STOMACH)` not already do that? There's also:
`if(istype(stomach))` which I thought also made sure they had their stomach. I'd assumed that's how it happened, if not I'll be good with doing something

Reply:
i agree

Reply:
Should be good, I also added them to light and cell power draining